src/dbwrapper.h

```diff
@@ -219,6 +223,12 @@ class CDBWrapper
 
     std::vector<unsigned char> CreateObfuscateKey() const;
 
+    //! path to filesystem storage
+    const fs::path m_path;
+
+    //! whether or not the database resides in memory
+    bool m_is_memory;
+
```

Comment:
nit in 75bb0c54b383e434695bf6bc1d60ef5bcd83cd70:

Can combine both into

```cpp
    //! path to filesystem storage or std::nullopt if database resides in memory
    std::optional<fs::path> m_path;

Reply:
I think @ryanofsky brought this up earlier; while I agree it would be a good change, it will broaden the scope of this PR quite a bit because all uses of CDBWrapper will have to change, since `path` is currently a required argument. PR is already large, so this could be a follow-up.